pages/queue.js

```diff
@@ -46,7 +47,8 @@ class Queue extends React.Component {
   }
 
   render() {
-    const { isFetching, hasQueue } = this.props
+    const { isFetching, hasQueue, queueInfo } = this.props
+    const location = this.props.queueInfo.location || 'Not Available'
```

Comment:
`queueInfo` might not be defined while the page is still loading (try refreshing the queue page with your code as-is). Avoid accessing any properties until after the various `isFetching` checks.